Tools/Scripts/webkitpy/style/checkers/jsonchecker.py

```diff
@@ -418,3 +423,107 @@ def check_codegen_properties(self, property_name, codegen_properties):
                 return
 
             keys_and_validators[key](property_name, 'codegen-properties', key, value)
+
+
+class JSONImportExpectationsChecker(JSONChecker):
+    """Processes the import-expectations.json lines"""
+
+    def check(self, lines):
+        super(JSONImportExpectationsChecker, self).check(lines)
+
+        try:
+            expectations = json.loads("\n".join(lines) + "\n")
+        except ValueError:
+            # Skip the rest, the parent class will have logged for this
+            return
+
+        if not isinstance(expectations, dict):
+            self._handle_style_error(
+                0, "json/syntax", 5, "The top-level data must be an object"
+            )
+            return
+
+        parsed_expectations = {}
+
+        for key, value in expectations.items():
+            valid = True
+
+            # This is an assert because JSON requires it, and thus should be infallible.
+            assert isinstance(key, six.text_type)
+
+            if not key == "web-platform-tests" and not key.startswith(
+                "web-platform-tests/"
+            ):
+                self._handle_style_error(
+                    0,
```

Comment:
Not clear that '0' is the right choice here. This should really be the line number of the error. You might consider searching for `'"{}"'.format(key)` in the original file to assign a line, somewhat like we do in `line_number_from_json_exception`

Reply:
`line_number_from_json_exception` is looking at the `json.JSONDecodeError` (or `ValueError`, prior to the former's existence), rather than searching the whole file. And really when `json.JSONDecodeError` exists it should just read from that object rather than parsing the message.

`JSONFeaturesChecker` and `JSONCSSPropertiesChecker` both use zero for everything, specifically because we can't accurately attribute errors.

I guess we can (fallibly) guess? But e.g. if someone does something like:

```json
{
    "foo/bar": "web-platform-tests/a/b/c",
    "web-platform-tests/a/b": "import"
    "web-pl\u0061tform-tests/a/b/c": "import"
}
```

…we'll almost certainly blame the wrong line.

Reply:
Well, I guess we can find all strings in JSON easily enough and parse them and attribute them to lines. It's kinda messy, but I guess it works in general.